Update CAPA helm charts build

serngawy opened the pull request.

marek-veber commented on charts/cluster-api-provider-aws/values.yaml:
```diff
@@ -1,6 +1,8 @@
+awsIAMRole: ""
+ocmToken: ""
 manager:
+  cmd: /bin/cluster-api-provider-aws-controller-manager
```
I have no problem with it, but:
 * Why introduce new variables?
 * Where is the boundary for deciding when to use a variable and when not to?

serngawy replied:
I added the command to be able use the upstream CAPA image , the binary in CAPA image registry.k8s.io/cluster-api-aws/cluster-api-aws-controller is ./manager

marek-veber reviewed (commented):
There was the final check:
```
helm template ./charts/"$PRJ" --include-crds |$YQ  ea '[.] | sort_by(.kind,.metadata.name) | .[] | splitDoc|sort_keys(..)' > /tmp/"$PRJ".yaml
```
1. it checks that the template is walid
2. the "sorted" output can be used to compare with the previous version (the "sorted" yaml templates files also reduce the size of commit changes)

serngawy commented:
> There was the final check:
> 
> ```
> helm template ./charts/"$PRJ" --include-crds |$YQ  ea '[.] | sort_by(.kind,.metadata.name) | .[] | splitDoc|sort_keys(..)' > /tmp/"$PRJ".yaml
> ```
> 
> 1. it checks that the template is walid
> 2. the "sorted" output can be used to compare with the previous version (the "sorted" yaml templates files also reduce the size of commit changes)
> 
> The final changes in the `helm template charts/cluster-api-provider-aws/ --include-crds |yq ea '[.] | sort_by(.kind,.metadata.name) | .[] | splitDoc|sort_keys(..)' > /tmp/cluster-api-provider-aws.yaml` are:
> 
Okay , will add it

> 1. some CRDs now have the additional annotation: `service.beta.openshift.io/inject-cabundle: "true"` Is it required for all capa CRDs ?
okay, will change to the CRDs only have the cert-manager annotation

> 2. feature-gates were changed:
>    
>    * EKSEnableIAM=false (it was true)
>    * EKSAllowAddRoles=false (it was true)
>    * MachinePool=false (it was true)
>    * AutoControllerIdentityCreator=true (it was false)
>    * TagUnmanagedNetworkResources=true (it was false)
will change it back , just a mistake 

> 3. `serviceAccountName: capa-controller-manager` -> `controller-manager`
I didn't change that , just used the output of default/kustomization.yaml

> 4. `imagePullPolicy: IfNotPresent` -> `imagePullPolicy: Always`
the default value for `imagePullPolicy` is `IfNotPresent` 

> 5. new ENV option OCM_TOKEN
Yes, it is required to create ROSA-HCP and once Milan work is done we will add a new secret to the templates 

> 6. image 4.17 -> 4.18
Yes, this what we are tageting

> 7. removed annotation for the Service: `service.beta.openshift.io/serving-cert-secret-name: capa-webhook-service-cert`
okay, will add it , I need to test it again

serngawy commented:
I update the PR addressing all your comments AND as we discussed about the --featureGate,  I added your func [here](https://github.com/serngawy/cluster-api-installer/blob/capa/scripts/build.sh#L24) with TODO to modify it.

marek-veber commented:
When I start `make build` than there is difference in generated files:
* `capa-controller-manager` -> `controller-manager`
* `capa-system` -> `system`
